### PR TITLE
Add MaxConsecutiveFalse, MaxConsecutiveNegatives, MaxConsecutivePositives,  MaxConsecutiveTrue, MaxConsecutiveZeros, NumConsecutiveGreaterMean, NumConsecutiveLessMean

### PR DESCRIPTION
**Pull request by gsheni**

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ Future Release
 ==============
     * Enhancements
         * Add ``TimeSinceLastFalse``, ``TimeSinceLastMax``, ``TimeSinceLastMin``, and ``TimeSinceLastTrue`` primitives (:pr:`2418`)
+        * Add MaxConsecutiveFalse, MaxConsecutiveNegatives, MaxConsecutivePositives, MaxConsecutiveTrue, MaxConsecutiveZeros, NumConsecutiveGreaterMean, NumConsecutiveLessMean (:pr:`2420`)
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:10]: nit: Include class names in double back-tics for consistent formatting.

File: featuretools/primitives/standard/aggregation/max_consecutive_false.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        `NaN` values can be ignored with the `skipna` parameter
+
+        >>> max_consecutive_false_skipna = MaxConsecutiveFalse(skipna=False)
+        >>> max_consecutive_false_skipna([False, None, False, False, True, False])
+        2
+    """
+
+    name = "max_consecutive_false"
+    input_types = [ColumnSchema(logical_type=Boolean)]
```
Comment (thehomebrewnerd) [inline featuretools/primitives/standard/aggregation/max_consecutive_false.py:26]: Update to work with BooleanNullable? Not sure if things will work with `pd.NA` values present, but we should make sure to test if we add this input type.

File: featuretools/primitives/standard/aggregation/max_consecutive_true.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        `NaN` values can be included with the `skipna` parameter
+
+        >>> max_consecutive_true_skipna = MaxConsecutiveTrue(skipna=False)
+        >>> max_consecutive_true_skipna([True, None, True, True, True, False])
+        3
+    """
+
+    name = "max_consecutive_true"
+    input_types = [ColumnSchema(logical_type=Boolean)]
```
Comment (thehomebrewnerd) [inline featuretools/primitives/standard/aggregation/max_consecutive_true.py:26]: Add BooleanNullable?

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        array = pd.Series([True, True, True, True])
+        assert primitive_func(array) == 0
+
+    def test_all_false(self):
+        primitive_instance = self.primitive()
+        primitive_func = primitive_instance.get_function()
+        array = pd.Series([False, False, False])
+        assert primitive_func(array) == 3
+
+    def test_with_nan(self):
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:34]: Maybe we are already testing with nan even though we would never get this with the `Boolean` logical type.

File: docs/source/release_notes.rst (hunk 1 above)
Comment (gsheni) [reply docs/source/release_notes.rst:10]: Fixed

File: featuretools/primitives/standard/aggregation/max_consecutive_false.py (hunk 2 above)
Comment (gsheni) [reply featuretools/primitives/standard/aggregation/max_consecutive_false.py:26]: No, for now let's keep this just as Boolean. The function doesn't work with BooleanNullable (and pd.NA).

File: featuretools/primitives/standard/aggregation/max_consecutive_true.py (hunk 3 above)
Comment (gsheni) [reply featuretools/primitives/standard/aggregation/max_consecutive_true.py:26]: See above

File: featuretools/tests/primitive_tests/test_max_consecutive.py (hunk 4 above)
Comment (gsheni) [reply featuretools/tests/primitive_tests/test_max_consecutive.py:34]: Removed all the nan test cases as they don't make sense.

File: featuretools/primitives/standard/aggregation/max_consecutive_positives.py [hunk 5]
```diff
@@ -23,7 +23,10 @@ class MaxConsecutivePositives(AggregationPrimitive):
     """
 
     name = "max_consecutive_positives"
-    input_types = [ColumnSchema(semantic_tags={"numeric"})]
+    input_types = [
+        [ColumnSchema(logical_type=Integer)],
```
Comment (thehomebrewnerd) [inline featuretools/primitives/standard/aggregation/max_consecutive_positives.py:27]: So I guess this means this also doesn't work with the IntegerNullable logical type? We should create an issue to update these to work with all numeric types. It's not common for us to have primitives that work on only a subset of numeric types...

File: featuretools/primitives/standard/aggregation/max_consecutive_zeros.py [hunk 6]
```diff
@@ -23,6 +23,10 @@ class MaxConsecutiveZeros(AggregationPrimitive):
     """
 
     name = "max_consecutive_zeros"
+    input_types = [
+        [ColumnSchema(logical_type=Integer)],
+        [ColumnSchema(logical_type=Double)],
+    ]
     input_types = [ColumnSchema(semantic_tags={"numeric"})]
```
Comment (thehomebrewnerd) [inline featuretools/primitives/standard/aggregation/max_consecutive_zeros.py:30]: Input types is defined twice on L26 and L30.

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 7]
```diff
@@ -11,230 +12,211 @@
...
 
+@pytest.mark.parametrize("dtype", ["float64", "int64"])
 class TestMaxConsecutiveNegatives:
-    primitive = MaxConsecutiveNegatives
-
-    def test_regular(self):
-        primitive_instance = self.primitive()
+    def test_regular(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:58]: floats -> nans

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 8]
```diff
@@ -11,230 +12,211 @@
...
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([1, -1, 2, 4, -5])
+        array = pd.Series([1, -1, 2, 4, -5], dtype=dtype)
         assert primitive_func(array) == 1
 
-    def test_all_float(self):
-        primitive_instance = self.primitive()
+    def test_all_float(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:72]: floats -> nans

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 9]
```diff
@@ -11,230 +12,211 @@
...
-        primitive_instance = self.primitive()
-        primitive_func = primitive_instance.get_function()
-        array = pd.Series([1, -1, 2, 4, -5])
-        assert primitive_func(array) == 2
-
-    def test_all_float(self):
-        primitive_instance = self.primitive()
+    def test_regular(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:115]: floats -> nans

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 10]
```diff
@@ -11,230 +12,211 @@
...
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([1, np.nan, 2, -3])
+        array = pd.Series([1, -1, 2, 4, -5], dtype=dtype)
         assert primitive_func(array) == 2
 
-    def test_with_nan_skipna(self):
-        primitive_instance = self.primitive(skipna=False)
+    def test_all_float(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:129]: floats -> nans

File: featuretools/primitives/standard/aggregation/max_consecutive_positives.py (hunk 5 above)
Comment (gsheni) [reply featuretools/primitives/standard/aggregation/max_consecutive_positives.py:27]: Created
https://github.com/alteryx/featuretools/issues/2422
https://github.com/alteryx/featuretools/pull/2420

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 11]
```diff
@@ -11,230 +12,211 @@
...
 
+@pytest.mark.parametrize("dtype", ["float64", "int64"])
 class TestMaxConsecutiveZeros:
-    primitive = MaxConsecutiveZeros
-
-    def test_regular(self):
-        primitive_instance = self.primitive()
+    def test_regular(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:172]: floats -> nans

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 12]
```diff
@@ -11,230 +12,211 @@
...
         primitive_func = primitive_instance.get_function()
-        array = pd.Series([1, -1, 0, 0, -5])
+        array = pd.Series([1, -1, 0, 0, -5], dtype=dtype)
         assert primitive_func(array) == 2
 
-    def test_all_float(self):
-        primitive_instance = self.primitive()
+    def test_all_float(self, dtype):
+        if dtype == "int64":
+            pytest.skip("floats not supported in int64")
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:186]: floats -> nans

**Review by thehomebrewnerd — COMMENTED** [review]
Looks like we have two new uncovered lines with these updates. Might be nice to add a test to cover those if it's not a big lift.

File: featuretools/tests/primitive_tests/test_max_consecutive.py (hunk 12 above)
Comment (gsheni) [reply featuretools/tests/primitive_tests/test_max_consecutive.py]: No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556

File: featuretools/tests/primitive_tests/test_max_consecutive.py (hunk 11 above)
Comment (gsheni) [reply featuretools/tests/primitive_tests/test_max_consecutive.py]: No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556

File: featuretools/tests/primitive_tests/test_max_consecutive.py (hunk 10 above)
Comment (gsheni) [reply featuretools/tests/primitive_tests/test_max_consecutive.py]: No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556

File: featuretools/tests/primitive_tests/test_max_consecutive.py (hunk 9 above)
Comment (gsheni) [reply featuretools/tests/primitive_tests/test_max_consecutive.py]: No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556

File: featuretools/tests/primitive_tests/test_max_consecutive.py (hunk 8 above)
Comment (gsheni) [reply featuretools/tests/primitive_tests/test_max_consecutive.py]: No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556

File: featuretools/tests/primitive_tests/test_max_consecutive.py (hunk 7 above)
Comment (gsheni) [reply featuretools/tests/primitive_tests/test_max_consecutive.py]: No this was intentional message. I made a better one
https://github.com/alteryx/featuretools/pull/2420/commits/a014517e40ce6dde0475b6c242bfe4566b183556

File: featuretools/primitives/standard/aggregation/max_consecutive_zeros.py (hunk 6 above)
Comment (gsheni) [reply featuretools/primitives/standard/aggregation/max_consecutive_zeros.py]: Removed

File: featuretools/tests/primitive_tests/test_max_consecutive.py (hunk 7 above)
Comment (thehomebrewnerd) [reply featuretools/tests/primitive_tests/test_max_consecutive.py]: Oh, I see. I didn't follow what was being tested closely enough.

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 13]
```diff
@@ -69,7 +69,7 @@ def test_all_int(self, dtype):
 
     def test_all_float(self, dtype):
         if dtype == "int64":
-            pytest.skip("floats not supported in int64")
+            pytest.skip("test array contains floats which are not supported int64")
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:72]: Nitpick: the floats used in this test could all be represented as integers. Should we change?

File: featuretools/tests/primitive_tests/test_max_consecutive.py [hunk 14]
```diff
@@ -126,7 +126,7 @@ def test_all_int(self, dtype):
...
```
Comment (thehomebrewnerd) [inline featuretools/tests/primitive_tests/test_max_consecutive.py:129]: Nitpick: the floats used in this test could all be represented as integers. Should we change?

**Review by thehomebrewnerd — APPROVED** [review]
Small nitpicks, but not blocking

**Pull request merged**